README.md

```diff
@@ -1,62 +1,272 @@
+
+To accomplish this you can call `styled` as a function and pass in the previous component. Then you style that like any other styled-component, except it overrides the styles you pass in and keeps the not overridden ones around:
+
+```JSX
+// Tomatobutton.js
+
+import React from 'react';
+import styled from 'styled-components';
+
+import Button from './Button';
+
+const TomatoButton = styled(Button)`
+color: tomato;
+border-color: tomato;
+`;
+
+export default TomatoButton;
+```
+
+<div align="center">
+  <a href="http://www.webpackbin.com/VJZQkBU6Z">
+    <img alt="Screenshot of the above code ran in a browser" src="http://imgur.com/LZZ3h5i.jpg" />
+    <div><em>Live demo</em></div>
+  </a>
+</div>
+
+Instead of copy and pasting the styles or factoring out the CSS into a separate function we've now reused a part of the styles. By leveraging components as our low-level styling construct, our codebase becomes a lot clearer!
+
+> You can also pass tag names into the `styled()` call, like so: `styled('div')`. In fact, the styled.tagname helpers are just aliases of `styled('tagname')`!
+
+#### Third-party components
+
+The above also works perfectly for styling third-party components, like a `react-router` `<Link />`!
+
+```JS
+import styled from 'styled-components';
+import { Link } from 'react-router';
+
+const StyledLink = styled(Link)`
+  color: palevioletred;
+  display: block;
+  margin: 0.5em 0;
+  font-family: Helvetica, Arial, sans-serif;
+
+  &:hover {
+    text-decoration: underline;
   }
-`
-
-const Main = styled.section`
-  max-width: 800px;
-  margin: 0 auto;
-`
-
-const P = styled.p`
-  line-height: 1.4;
-  margin-bottom: 1.5rem;
-`
-
-export default () => (
-  <div>
-    <Header>
-      <h1>Styled Components</h1>
-      <h2>They're so rad!</h2>
-    </Header>
-    <Main>
-      <P>
-        Separate those concerns! JSX is for structure, styled.tagName for styling!
-      </P>
-      <P>
-        Just gotta use component names that start with an Uppercase letter and you're good to get started!
-      </P>
-    </Main>
-  </div>
-)
-```
-
-By Glen Maddern and Max Stoiber.
-
-With thanks to Charlie Somerville & lots of others.
+`;
+```
+
+```JSX
+<Link to="/">Standard, unstyled Link</Link>
+<StyledLink to="/">This Link is styled!</StyledLink>
+```
+
+<div align="center">
+  <a href="http://www.webpackbin.com/41PeBHU6-">
+    <img alt="Screenshot of the above code ran in a browser" src="http://imgur.com/JJw4MdX.jpg" />
+    <div><em>Live demo</em></div>
+  </a>
+</div>
+
+### Keyframes
+
+Keyframes in CSS don't make sense to be scoped to a single component, they are meant to be global. This is why we export a `keyframes` helper which will generate a unique name for your keyframes. You can then use that unique name throughout your app.
+
+This way, you get all the benefits of using JavaScript, are avoiding name clashes and get your keyframes like always:
+
+```JS
+import { keyframes } from 'styled-components';
+
+const keyframes = {
```

Comment:
having the word `keyframes` here is confusing because it's the same name you imported. Just `export const fadeIn` is better